library/src/scala/quoted/Quotes.scala

```diff
@@ -2453,10 +2452,18 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
       extension (self: MatchType):
         def bound: TypeRepr
         def scrutinee: TypeRepr
-        def cases: List[TypeRepr]
+        def cases: List[MatchTypeCase]
       end extension
     end MatchTypeMethods
 
+    type MatchTypeCase <: TypeRepr
```

Comment:
We should keep the abstractions aligned with what is in TASTy and not add new ones.